### PR TITLE
Add floatingip data source

### PR DESCRIPTION
**Pull request by rasta-rocket**

### CODE REVIEW

**Comment by rasta-rocket** [comment]
Hi @jtopjian 

I got a TIMED_OUT on terraform-provider-openstack-acceptance-test, can you provide me some help please ?

Thx :)

**Comment by jtopjian** [comment]
recheck

**Comment by jtopjian** [comment]
@rasta-rocket The tests are re-running now.

I manually ran the test and got the following error:

```
--- FAIL: TestAccOpenStackNetworkingFloatingIPV2DataSource_address (21.53s)
        testing.go:518: Step 1 error: Check failed: Check 4/4 error: data.openstack_networking_floatingip_v2.fip: Attribute 'status' expected "ACTIVE", got "DOWN"
```

File: website/docs/d/networking_floatingip_v2.html.markdown [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+layout: "openstack"
+page_title: "OpenStack: openstack_networking_floatingip_v2"
+sidebar_current: "docs-openstack-datasource-networking-floatingip-v2"
+description: |-
+  Get information on an OpenStack Subnet.
```
Comment (jtopjian) [inline website/docs/d/networking_floatingip_v2.html.markdown]: Get information on an OpenStack Floating IP

File: website/docs/d/networking_floatingip_v2.html.markdown [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+---
+
+# openstack\_networking\_floatingip\_v2
+
+Use this data source to get the ID of an available OpenStack floatingip.
```
Comment (jtopjian) [inline website/docs/d/networking_floatingip_v2.html.markdown]: floatingip => Floating IP

File: website/docs/d/networking_floatingip_v2.html.markdown [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+}
+```
+
+## Argument Reference
+
+* `region` - (Optional) The region in which to obtain the V2 Neutron client.
+  A Neutron client is needed to retrieve floating IP ids. If omitted, the
+  `region` argument of the provider is used.
+
+* `address` - (Optional) The IP version of the floating IP (either 4 or 6).
```
Comment (jtopjian) [inline website/docs/d/networking_floatingip_v2.html.markdown]: This should probably read something like "The address of the floating IP"

File: website/docs/d/networking_floatingip_v2.html.markdown [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+* `pool` - (Optional) The name of the floating IP's gateway.
```
Comment (jtopjian) [inline website/docs/d/networking_floatingip_v2.html.markdown]: This should probably read "The pool of the floating IP"

File: website/docs/d/networking_floatingip_v2.html.markdown [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+* `port_id` - (Optional) The ID of the port the floating IP is attached.
+
+* `fixed_ip` - (Optional) The specific IP address of the internal port which should be associated with the floating IP.
+
+* `tenant_id` - (Optional) The owner of the floatingIP.
```
Comment (jtopjian) [inline website/docs/d/networking_floatingip_v2.html.markdown]: floating ip => floating IP

File: website/docs/d/networking_floatingip_v2.html.markdown [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+## Attributes Reference
+
+`id` is set to the ID of the found floatingip. In addition, the following attributes
+are exported:
+
+* `status` - status of the floatingip (ACTIVE/DOWN).
```
Comment (jtopjian) [inline website/docs/d/networking_floatingip_v2.html.markdown]: floatingip => floating IP

File: openstack/data_source_openstack_networking_floatingip_v2.go [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+		listOpts.PortID = v.(string)
+	}
+
+	if v, ok := d.GetOk("fixed_ip"); ok {
+		listOpts.FixedIP = v.(string)
+	}
+
+	pages, err := floatingips.List(networkingClient, listOpts).AllPages()
+	if err != nil {
+		return err
```
Comment (jtopjian) [inline openstack/data_source_openstack_networking_floatingip_v2.go]:
```go
return fmt.Errorf("Unable to list Floating IPs: %s", err)
```

**Comment by rasta-rocket** [comment]
@jtopjian For acceptance test, it's stranged ... because on my own implementation FIPs which are not associated with a device are shown as ACTIVE, but it's not impossible that my neutron implementation is a bit buggy :D (I'm using opencontrail)
Anyway I have removed the check of the status value (not sure it is very useful here).
Thx for the comments and the review ;)

**Comment by jtopjian** [comment]
> I'm using opencontrail

Yes, we've seen some odd behavior from opencontrail-based environments in the past. :)

**Pull request merged**